diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -377,9 +377,20 @@ int DifferentialFunctionComparator::cmpOperations(
             auto CmpL = dyn_cast<CmpInst>(L);
             auto CmpR = dyn_cast<CmpInst>(R);
 
+            // Only apply inverse condition pattern if the operands are of the
+            // same type, i.e. cmpOperations returned non-0 because of the
+            // values rather than the types.
+            bool typesMatch = true;
+            for (unsigned i = 0, e = CmpL->getNumOperands(); i != e; ++i) {
+                if (cmpTypes(L->getOperand(i)->getType(),
+                             R->getOperand(i)->getType()))
+                    typesMatch = false;
+            }
```

Comment:
This looks like something that could be useful from other places. Maybe extract into a function (e.g. `cmpOperandTypes`)?

Reply:
Good idea, done.